app/src/main/java/org/jellyfin/androidtv/ui/search/SearchViewModel.kt

```diff
@@ -0,0 +1,70 @@
+package org.jellyfin.androidtv.ui.search
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.util.apiclient.waitForSuccess
+import org.jellyfin.sdk.model.api.BaseItemKind
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+
+class SearchViewModel : ViewModel() {
+
+	private var searchJob: Job? = null
+
+	private var previousQuery: String? = null
+
+	private val _searchResultsFlow = MutableStateFlow<List<SearchHintGroup>>(emptyList())
+	val searchResultsFlow = _searchResultsFlow.asStateFlow()
+
+	fun search(query: String) = searchDebounced(query, 0.milliseconds)
+
+	fun searchDebounced(query: String, debounceMs: Duration = debounceDuration): Boolean {
+		if (query == previousQuery) return false
+		previousQuery = query
+
+		if (query.isBlank()) {
+			_searchResultsFlow.value = emptyList()
+			return true
+		}
+
+		searchJob?.cancel()
+		searchJob = viewModelScope.launch {
+			delay(debounceMs)
+			val deferredResults = groups.map { (stringRes, itemKinds) ->
+				async {
+					val result = SearchRepository.search(query, itemKinds).waitForSuccess()
+					val items = result?.content?.items ?: emptyList()
+					SearchHintGroup(stringRes, items)
+				}
+			}
+			val results = deferredResults.awaitAll()
```

Comment:
This will run all search calls in parallel, which is what the web client appears to be doing